selfie.c

```diff
@@ -566,6 +566,97 @@ initParser() {
 
 void emitLeftShiftBy(int b);
 void emitMainEntry();
+// *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~
+// -----------------------------------------------------------------
+// ---------------------     O S   ---------------------------------
+// -----------------------------------------------------------------
+// *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~ *~*~
+
+
+// -----------------------------------------------------------------
+// ------------------------- Assignment 0 Linked List --------------
+// -----------------------------------------------------------------
+
+// Eg.: Size 2
+// list
+// We add 2 to the given size
+// for the pointer 
+// The size variable defines the 
+// entrys you need for data
+// +---------------+
+// | Entry         |
+// +---------------+
+// | Entry 2       |
+// +---------------+
+// | Pr next 0     |
+// +---------------+
+// | Pr Prev 0     |
+// +---------------+
+   
+int createLList(int size){
+   int *list;
+   int *pr;
+   list = malloc((size + 2) * 4);
+   pr = list;
+   pr = pr + size;
+   *pr = 0;
+   pr = pr + 1;
+   *pr = 0;
+   return list;
+}
+
+int addNodeToLList(int size, int list){
```

Comment:
see comment on data layout. Once you've got the convention that prev and next are the first 2 words in any node "structure", you get much simpler code for insertion and removal of nodes.